### DISCUSSION TITLE
Overriding dependencies with liveValue only

### DISCUSSION THREAD

**Discussion by Budyn** [pointfreeco/swift-dependencies#205] [Q&A]
Hello! 

I'm migrating the Dependencies package to 1.2, so the new overriding dependecies technique can be used. 

My project has protocol oriented dependencies.

In order to use this syntax:
`@Dependency(FetchServiceKey.self) var fetchService`

I have to include the **testValue** inside the DependencyKey object:

```swift
protocol FetchService {
    // ...
}

enum FetchServiceKey: DependencyKey {
    static let liveValue: EntryService = LiveEntryService()

    // It has to be present in order to unlock the new overriding method
    static let testValue = liveValue
}

struct LiveFetchService: FetchService {
    // ...
}
```

By commenting out the testValue from the DependencyKey, I can't build the project and have a warning:

```
Variable 'fetchService' inferred to have type 'FetchServiceKey', which is an enum with no cases
Add an explicit type annotation to silence this warning
```

Do we need to include the testValue each time for this overriding method?

**Comment by mbrandonw** [comment]
Hi @Budyn, can you provide a complete, compiling sample that demonstrates the problem? This compiles just fine with no warnings:

```swift
protocol FetchService {}
struct LiveFetchService: FetchService {}
enum FetchServiceKey: DependencyKey {
  static let liveValue: any FetchService = LiveFetchService()
}
func foo() {
  @Dependency(FetchServiceKey.self) var fetchService
}
```

**Reply by Budyn** [reply]
Hello @mbrandonw, thank you for the quick response. 

This is the extended example, that invokes the mentioned earlier warning:

```swift
protocol FetchService {
    func doWork()
}

struct LiveFetchService: FetchService {
    func doWork() {}
}

enum FetchServiceKey: DependencyKey {
    static let liveValue: any FetchService = LiveFetchService()
}

protocol ClientService {
    func doWork()
}

struct LiveClientService: ClientService {

    @Dependency(FetchServiceKey.self) var fetchService

    func requestWork() {
        fetchService.doWork()
    }
}

enum ClientServiceKey: DependencyKey {
    static let liveValue: any ClientService = LiveClientService()
}

func foo() {
    @Dependency(ClientServiceKey.self) var clientService
    clientService.doWork()
}
```

**Reply by mbrandonw** [reply]
Hi @Budyn, that code does not compile because you use `requestWork` instead of `doWork` in `LiveClientService`, but once you make that change it compiles just fine with no warnings.
